src/nwb_conversion_tools/tools/neo/neo.py

```diff
@@ -0,0 +1,602 @@
+
+    assert icephys_experiment_type in ["voltage_clamp", "current_clamp", "izero"], (
+        f"'icephys_experiment_type' should be 'voltage_clamp', 'current_clamp' or 'izero', but received value "
+        f"{icephys_experiment_type}"
+    )
+
+    # Check and auto-create electrodes, in case they don't exist yet in nwbfile
+    if len(nwbfile.icephys_electrodes) == 0:
+        warnings.warn(
+            "When adding Icephys Recording, no Icephys Electrodes were found on nwbfile. Creating Electrodes now..."
+        )
+        add_icephys_electrode(
+            neo_reader=neo_reader,
+            nwbfile=nwbfile,
+            metadata=metadata,
+        )
+
+    if getattr(nwbfile, "intracellular_recordings", None):
+        ri = max(nwbfile.intracellular_recordings["responses"].index)
+    else:
+        ri = -1
+    if getattr(nwbfile, "icephys_simultaneous_recordings", None):
+        simultaneous_recordings_offset = len(nwbfile.icephys_simultaneous_recordings)
+    else:
+        simultaneous_recordings_offset = 0
+
+    if getattr(nwbfile, "icephys_sequential_recordings", None):
+        sessions_offset = len(nwbfile.icephys_sequential_recordings)
+    else:
+        sessions_offset = 0
+
+    relative_session_start_time = metadata["Icephys"]["Sessions"][sessions_offset]["relative_session_start_time"]
+    session_stimulus_type = metadata["Icephys"]["Sessions"][sessions_offset]["stimulus_type"]
+
+    # Sequential icephys recordings
+    simultaneous_recordings = list()
+    for si in range(n_segments):
+        # Parallel icephys recordings
+        recordings = list()
+        for ei, electrode in enumerate(
+            list(nwbfile.icephys_electrodes.values())[: len(neo_reader.header["signal_channels"]["units"])]
+        ):
+            if ei in skip_electrodes:
+                continue
+            # Starting time is the signal starting time within .abf file + time
+            # relative to first session (first .abf file)
+            ri += 1
+            starting_time = neo_reader.get_signal_t_start(block_index=0, seg_index=si)
+            starting_time = starting_time + relative_session_start_time
+
+            sampling_rate = neo_reader.get_signal_sampling_rate()
+            response_unit = neo_reader.header["signal_channels"]["units"][ei]
+            response_conversion = get_conversion_from_unit(unit=response_unit)
+            response_gain = neo_reader.header["signal_channels"]["gain"][ei]
+            response_name = f"{icephys_experiment_type}-response-{si + 1 + simultaneous_recordings_offset:02}-ch-{ei}"
+
+            response = response_classes[icephys_experiment_type](
+                name=response_name,
+                description=f"Response to: {session_stimulus_type}",
+                electrode=electrode,
+                data=neo_reader.get_analogsignal_chunk(block_index=0, seg_index=si, channel_indexes=ei),
```

Comment:
Skipping iterative write is fine as long as we can guarantee that in practice, such files are never more than 16GB.

We should enable compression, though, just wrap in an `H5DataIO` and pass the compression options at https://github.com/catalystneuro/nwb-conversion-tools/pull/502/files#diff-c2e3bb3498bea80fe7bf252c1c7cda1365a1aa1fb0f4ba8fe3f5fece8d16e0bfR378-R379

Reply:
added compression